homeassistant/components/edl21/sensor.py

```diff
@@ -58,196 +58,160 @@ class EDL21:
     """EDL21 handles telegrams sent by a compatible smart meter."""
 
     # OBIS format: A-B:C.D.E*F
-    _OBIS_SENSOR_ENTITY_DESCRIPTIONS = {
+    _OBIS_SENSOR_PROPERTIES = {
         # A=1: Electricity
         # C=0: General purpose objects
         # D=0: Free ID-numbers for utilities
-        "1-0:0.0.9*255": SensorEntityDescription(
```

Comment:
Why did you revert the use of entity descriptions? That is actually the preferred method to use.

Reply:
I thought it's better to create the entity descriptions only for the sensors that are actually supported by the respective smart meter (see [line 268](https://github.com/home-assistant/core/pull/56968/files/18d24451bfd51e79fab431ba7ae835c57cea48ad..e31dbb742bb3214b0ffd4b9a328df2d5b2b95030#diff-5e4e812555e1b089aa137084f7dac49e1f457f0a8addc1eb46299b01b2c40220R268)).
If you think this is premature optimization, I will change it "back" to 
```
"1-0:96.1.0*255": SensorEntityDescription(
            key="1-0:96.1.0*255", name="Metering point ID 1"
        ),
```

Reply:
That is not how SensorEntitiesDescription are meant to be used.

Instead replace your current mapping with a tuple of SensorEntityDescriptions

Reply:
I can replace the mapping with a tuple, no problem. But, then I would need to search within this tuple for the matching SensorEntityDescriptions instead of taking it directly from the mapping as currently done [here](https://github.com/home-assistant/core/blob/e31dbb742bb3214b0ffd4b9a328df2d5b2b95030/homeassistant/components/edl21/sensor.py#L263).

Just for my clarification, as far as I understand EDL21 meters and this code (@mtdcr Please correct me if I'm wrong):
- EDL21 meters support a set of metrics each identified by an OBIS code (e.g. "1-0:96.1.0*255")
- You can not know which metrics are supported by your EDL21 meter until your receive the first SML telegram. Each SML telegram is basically a list of key-value pairs: OBIS code and the respective metric value. (The smart meter I have only supports 7 of the 27 metrics listed in the code.)
- So the code receives the first SML telegram and for each OBIS code found in the telegram it looks up the matching SensorEntitiesDescription in the mentioned mapping and creates a SensorEntity using this SensorEntitiesDescription.
- For each further SML telegram the code "just" updates the states of the respective SensorEntities via _async_write_ha_state_.

If this behaviour contradicts architectural patterns/best practices in HomeAssistant, I am happy to refactor it.
So imho the dictonary is the better approach here (unless of cause we just want to create SensorEntities for all possible OBIS code, even when they are not supported by the smart meter).

Reply:
We generally handle that as follows:

```py
SENSORS: tuple[SensorEntityDescription, ...] = (
    SensorEntityDescription(
    ),
    SensorEntityDescription(
    ),
)

SENSOR_KEYS: list[str] = [desc.key for desc in SENSOR_TYPES]
```

Reply:
Thanks for the hint (and for taking the time). I changed the code accordingly.